packages/augur-core/source/contracts/trading/CreateOrder.sol

```diff
@@ -107,4 +112,79 @@ contract CreateOrder is Initializable, ReentrancyGuard {
 
         return _orders;
     }
+
+    //
+    // Private functions
+    //
+
+    function escrowFunds(Order.Data memory _orderData) internal returns (bool) {
+        if (_orderData.orderType == Order.Types.Ask) {
+            return escrowFundsForAsk(_orderData);
+        } else if (_orderData.orderType == Order.Types.Bid) {
+            return escrowFundsForBid(_orderData);
+        }
+    }
+
+    function escrowFundsForBid(Order.Data memory _orderData) private returns (bool) {
+        require(_orderData.moneyEscrowed == 0, "Order.escrowFundsForBid: New order had money escrowed. This should not be possible");
+        require(_orderData.sharesEscrowed == 0, "Order.escrowFundsForBid: New order had shares escrowed. This should not be possible");
+        uint256 _attosharesToCover = _orderData.amount;
+        uint256 _numberOfShortOutcomes = _orderData.market.getNumberOfOutcomes() - 1;
+
+        uint256[] memory _shortOutcomes = new uint256[](_numberOfShortOutcomes);
+        uint256 _indexOutcome = 0;
+        for (uint256 _i = 0; _i < _numberOfShortOutcomes; _i++) {
+            if (_i == _orderData.outcome) {
+                _indexOutcome++;
+            }
+            _shortOutcomes[_i] = _indexOutcome;
+            _indexOutcome++;
+        }
+
+        // Figure out how many almost-complete-sets (just missing `outcome` share) the creator has
+        uint256 _attosharesHeld = _orderData.shareToken.lowestBalanceOfMarketOutcomes(_orderData.market, _shortOutcomes, _orderData.creator);
+
+        // Take shares into escrow if they have any almost-complete-sets
+        if (_attosharesHeld > 0) {
+            _orderData.sharesEscrowed = SafeMathUint256.min(_attosharesHeld, _attosharesToCover);
+            _attosharesToCover -= _orderData.sharesEscrowed;
+            uint256[] memory _values = new uint256[](_numberOfShortOutcomes);
+            for (uint256 _i = 0; _i < _numberOfShortOutcomes; _i++) {
+                _values[_i] = _orderData.sharesEscrowed;
+            }
+            _orderData.shareToken.unsafeBatchTransferFrom(_orderData.creator, address(_orderData.augurTrading), _orderData.shareToken.getTokenIds(_orderData.market, _shortOutcomes), _values);
+        }
+
+        // If not able to cover entire order with shares alone, then cover remaining with tokens
+        if (_attosharesToCover > 0) {
```

Comment:
Reading this code again after being away for a while makes me wonder how much simpler trading would be if we didn't allow mixed orders (part shares, part tokens).  I further wonder how frequently users _actually_ want to do position flip + raise like this.  It feels like we could just assert that orders are _either_ shares _or_ tokens and probably not negatively impact anyone, and likely also simplify a lot of code at the same time.

Reply:
Don't think its worth simplifying and removing functionality at this point